ext/spl/spl_directory.c

```diff
@@ -1935,20 +1935,30 @@ static zend_result spl_filesystem_file_read_ex(spl_filesystem_object *intern, bo
 	return SUCCESS;
 } /* }}} */
 
-static inline zend_result spl_filesystem_file_read(spl_filesystem_object *intern, bool silent)
+static inline zend_result spl_filesystem_file_read(spl_filesystem_object *intern, bool silent, bool csv)
 {
 	zend_long line_add = (intern->u.file.current_line) ? 1 : 0;
-	return spl_filesystem_file_read_ex(intern, silent, line_add);
+	return spl_filesystem_file_read_ex(intern, silent, line_add, csv);
+}
+
+static bool is_line_empty(spl_filesystem_object *intern)
```

Comment:
We already have ``spl_filesystem_file_is_empty_line()`` isn't this sufficient?

Reply:
I don't think that `spl_filesystem_file_is_empty_line()` is sufficient (and I doubt that it actually makes sense; IIRC, the `else if` body is dead code). And just checking that the `current_line_len == 0` is not enough to handle lines with only linebreak characters.

Reply:
Yeah, well can always be a follow-up job to get rid of this function and replace it by this new one.